src/common/react-novnc.js

```diff
@@ -86,14 +85,17 @@ export default class NoVnc extends Component {
     const rfb = this._rfb = new RFB({
       encrypt: isSecure,
       target: this.refs.canvas,
-      wsProtocols: [ 'chat' ],
+      wsProtocols: [ 'binary' ],
       onClipboard: onClipboardChange && ((_, text) => {
         onClipboardChange(text)
       }),
       onUpdateState: this._onUpdateState
     })
-
-    rfb.connect(formatUrl(url))
+    // remove heading slashes from the path
+    // it will be concatenated to a slash in the noVNC library (prefix + '/' + path) that would lead to: '//api/console/uid'
+    // then compared to a regex that only expects a single slash server side and miss the match.
+    const clippedPath = url.path.replace(/^\/+(.*)/, '$1')
+    rfb.connect(url.hostname, url.port, null, clippedPath)
```

Comment:
The former method signature was requesting a complete URL so Vates was directly creating the correct URL.
The new signature requires separate parameters for hostname, port, and path and noVNC does the concatenation to create the ws:// url.